panic: runtime error: invalid memory address or nil pointer dereference (26.17.0)

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### What happened?

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10067e8cc]

goroutine 20 [running]:
github.com/jandedobbeleer/oh-my-posh/src/segments.(*Terraform).setVersionFromTfFiles(0x140004aea80)
	github.com/jandedobbeleer/oh-my-posh/src/segments/terraform.go:95 +0x18c
github.com/jandedobbeleer/oh-my-posh/src/segments.(*Terraform).Enabled(0x140004aea80)
	github.com/jandedobbeleer/oh-my-posh/src/segments/terraform.go:47 +0x124
github.com/jandedobbeleer/oh-my-posh/src/config.(*Segment).Execute(0x140002a3c00, {0x100aa72f8, 0x14000050a80})
	github.com/jandedobbeleer/oh-my-posh/src/config/segment.go:128 +0x2a8
github.com/jandedobbeleer/oh-my-posh/src/prompt.(*Engine).writeSegmentsWithPool.func1()
	github.com/jandedobbeleer/oh-my-posh/src/prompt/segments.go:73 +0x8c
created by github.com/jandedobbeleer/oh-my-posh/src/prompt.(*Engine).writeSegmentsWithPool in goroutine 1
	github.com/jandedobbeleer/oh-my-posh/src/prompt/segments.go:70 +0x6c

<img width="809" height="444" alt="Image" src="https://github.com/user-attachments/assets/dd56e83e-1c64-4bc1-af12-87a83d96ffd0" />

### Theme

[`M365Princess-custom.omp.yaml`](https://github.com/SPONGE-JL/.mac-setup/blob/main/.zshrc-block/theme/M365Princess-custom.omp.yaml)

![sample](https://github.com/SPONGE-JL/.mac-setup/blob/main/.zshrc-block/theme/M365Princess-custom.omp.png?raw=true)

### What OS are you seeing the problem on?

macOS

### Which shell are you using?

zsh

### Log output

```Shell
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x1051b68cc]

goroutine 20 [running]:
github.com/jandedobbeleer/oh-my-posh/src/segments.(*Terraform).setVersionFromTfFiles(0x140001f5b40)
	github.com/jandedobbeleer/oh-my-posh/src/segments/terraform.go:95 +0x18c
github.com/jandedobbeleer/oh-my-posh/src/segments.(*Terraform).Enabled(0x140001f5b40)
	github.com/jandedobbeleer/oh-my-posh/src/segments/terraform.go:47 +0x124
github.com/jandedobbeleer/oh-my-posh/src/config.(*Segment).Execute(0x140002a3c00, {0x1055df2f8, 0x14000050ae0})
	github.com/jandedobbeleer/oh-my-posh/src/config/segment.go:128 +0x2a8
github.com/jandedobbeleer/oh-my-posh/src/prompt.(*Engine).writeSegmentsWithPool.func1()
	github.com/jandedobbeleer/oh-my-posh/src/prompt/segments.go:73 +0x8c
created by github.com/jandedobbeleer/oh-my-posh/src/prompt.(*Engine).writeSegmentsWithPool in goroutine 1
	github.com/jandedobbeleer/oh-my-posh/src/prompt/segments.go:70 +0x6c
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10467a8cc]

goroutine 29 [running]:
github.com/jandedobbeleer/oh-my-posh/src/segments.(*Terraform).setVersionFromTfFiles(0x140005a8fc0)
	github.com/jandedobbeleer/oh-my-posh/src/segments/terraform.go:95 +0x18c
github.com/jandedobbeleer/oh-my-posh/src/segments.(*Terraform).Enabled(0x140005a8fc0)
	github.com/jandedobbeleer/oh-my-posh/src/segments/terraform.go:47 +0x124
github.com/jandedobbeleer/oh-my-posh/src/config.(*Segment).Execute(0x1400031fc00, {0x104aa32f8, 0x140001809c0})
	github.com/jandedobbeleer/oh-my-posh/src/config/segment.go:128 +0x2a8
github.com/jandedobbeleer/oh-my-posh/src/prompt.(*Engine).writeSegmentsWithPool.func1()
	github.com/jandedobbeleer/oh-my-posh/src/prompt/segments.go:73 +0x8c
created by github.com/jandedobbeleer/oh-my-posh/src/prompt.(*Engine).writeSegmentsWithPool in goroutine 1
	github.com/jandedobbeleer/oh-my-posh/src/prompt/segments.go:70 +0x6c
```

--- a/src/segments/terraform.go
+++ b/src/segments/terraform.go
@@ -88,7 +88,7 @@ func (tf *Terraform) setVersionFromTfFiles() error {
 
 		var config TerraFormConfig
 		diags = gohcl.DecodeBody(hclFile.Body, nil, &config)
-		if diags != nil {
+		if diags != nil || config.Terraform == nil {
 			continue
 		}
 

